tests/integration/modifiers/will-destroy-test.js

```diff
@@ -43,4 +43,57 @@ module('Integration | Modifier | will-destroy', function (hooks) {
     // trigger destroy
     this.set('show', false);
   });
+
+  test('provides a useful error on install', async function (assert) {
+    assert.expect(1);
+
+    // Setup error capturing
+    setupOnerror(function (err) {
+      assert.equal(
+        err.toString(),
+        `TypeError: did-destroy expected a function, instead received "undefined"`
+      );
+    });
+
+    await render(hbs`
+      <div {{will-destroy this.nonExistentMethod}}></div>
+    `);
+
+    // Prevent double error on test teardown
+    this.set('nonExistentMethod', () => {});
+
+    // Reset error capturing
+    setupOnerror();
+  });
+
+  test('provides a useful error on destroy', async function (assert) {
+    assert.expect(1);
+
+    // Start with a valid function so that install works
+    this.set('nonExistentMethod', () => {});
+
+    // Setup error capturing
+    setupOnerror(function (err) {
+      assert.equal(
+        err.toString(),
+        `TypeError: did-destroy expected a function, instead received "undefined"`
+      );
+    });
+
+    this.set('show', true);
+    await render(hbs`
+      {{#if this.show}}
+        <div {{will-destroy this.nonExistentMethod}}></div>
+      {{/if}}
+    `);
+
+    // Remove the function to trigger an error on destroy
+    this.setProperties({
+      nonExistentMethod: undefined,
+      show: false,
+    });
+
```

Comment:
I’ve tried a couple of things to make sure this would work, but no luck so far… Curiously, it only fails with three older versions of Ember:
```
Scenario ember-lts-3.8: FAIL
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.8.0             │ 3.8.3                        │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-lts-3.12: FAIL
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.12.0            │ 3.12.4                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-lts-3.16: FAIL
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.16.0            │ 3.16.10                      │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘
```
Do you think we could potentially run these tests on all versions _except_ these?